src/Framework/LazyFormattedBuildEventArgs.cs

```diff
@@ -79,10 +79,9 @@ public LazyFormattedBuildEventArgs
         )
             : base(message, helpKeyword, senderName, eventTimestamp)
         {
-            arguments = messageArgs;
+            argumentsOrFormattedMessage = messageArgs;
             originalCultureName = CultureInfo.CurrentCulture.Name;
```

Comment:
This place was always bugging me. Why access the CurrentCulture twice? Are we sure we need to store both? Should we store just the culture name and recover culture on demand? Or store the culture and recover the name on demand?

Do we even need to store the culture in the first place? Are we ever in a situation where we have more than one culture?

Reply:
But perhaps we should revisit the culture in a separate PR, one thing at a time.

Reply:
Also, the culture name is only needed when converting args to string, In a case where we format message before serialization we always have Current CultureInfo, when we serialize we can use that to convert into culture aware string and `originalCultureName` does not need to become part of de/ser.

Reply:
We'll work on culture related fields separately. Tracked with #7015 (thank you!). I am resolving this conversation.